src/main/java/core/basesyntax/service/impl/WriterServiceImpl.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.WriterService;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class WriterServiceImpl implements WriterService {
+    @Override
+    public void writeToCsv(String generatedReport, String filePath) {
```

Comment:
Missing some checks before, like what if: 
`fileName == null
                || dataToWrite == null
                || !fileName.endsWith(".csv")`
                You are missing some checks + throwing exception in this case.